### PR TITLE
feat: ICS v5 Support (Base)

### LINKED ISSUE #112: ICS consumer chain generation
# TODO
- [x] Add / remove ICS as expected with tags
- [x] Add new 'staking' option. This feature is on by default but hidden
    - If ICS is used, then staking is disabled, removes all references for staking. \
    - Also remove Distr, Mint, etc.
- [x]  Integrate Interchaintest w/ simple ICS test ( https://github.com/strangelove-ventures/interchaintest/pull/1088 )
- [x] Add test_ics_node.sh instead of test_node.sh w/ Stide's provider hack
- [x] CI: Run test_ics_node.sh for ics chain (matrix?)
- [x] Speed up CI
- [ ] Unit Test (ref: stride, neutron)
- [x] see if we can abstract in setup.go so we can reuse standard test instead of new ones
    - [ ] do for all e2e test
- [x] Update to ICTest v8.3.0 + IBC  validation (from upstream ICT)
- [ ] Local Interchain proper generation (remove default)
- [ ] In UI, if ICS selected, deselect POA

# Test Basic
```
make install && rm -rf testics1/
spawn new testics1 --disable=poa,tf,globalfee,packetforward,ibc-ratelimit,cw,wasm-lc,ignite-cli

spawn new chain1 --disable=ics
spawn new chain2
```

### PR DESCRIPTION
**Pull request by reecepbcups**
ref #112

## Summary
Adds Interchain Security Consumer support

### CODE REVIEW

**Comment by Reecepbcups** [comment]
2f0e18bb44e378e86e38bc0b5552bd61224f5e32 is the legitist commit atm, after are optimizations which may not pan out

**Comment by reecepbcups** [comment]
Woo! Just on to auto generated UX testing. Then another PR can handle further multi-chain creation validation (mass matrix, ensure it go test / source is valid parsed)

Will still require the github new repo pushes for now. Would be nice to have a gh CLI cmd to delete those repos quickly (TODO)

File: simapp/interchaintest/ics_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	interchaintest "github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+// This tests Cosmos Interchain Security, spinning up a provider and a single consumer chain.
+func TestICS(t *testing.T) {
```
Comment (reecepbcups) [inline simapp/interchaintest/ics_test.go]: TODO: Update interchaintest to v8.3.0 here before merge (with ICS v8)

File: spawn/command.go [hunk 2]
```diff
@@ -25,4 +25,7 @@ func (cfg *NewChainConfig) GitInitNewProjectRepo() {
 	if err := ExecCommand("git", "commit", "-m", "initial commit", "--quiet"); err != nil {
 		cfg.Logger.Error("Error committing initial files", "err", err)
 	}
+	if err := os.Chdir(".."); err != nil {
```
Comment (reecepbcups) [inline spawn/command.go:28]: this cost me like an hour with tested. The cd broke future test cases :cry:

**Comment by reecepbcups** [comment]
PR so big, going to merge in and do some testing / validation before continuing to unit test & trying to generalized Interchaintest setup

**Pull request merged**